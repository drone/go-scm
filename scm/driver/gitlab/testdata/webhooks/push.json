{
  "object_kind": "push",
  "event_name": "push",
  "before": "9217710ce8c7e1eae7a5d1c45f6e43e1c769f866",
  "after": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
  "ref": "refs/heads/master",
  "checkout_sha": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
  "message": null,
  "user_id": 51764,
  "user_name": "Sid Sijbrandij",
  "user_username": "sytses",
  "user_email": "noreply@gitlab.com",
  "user_avatar": "https://secure.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87?s=80&d=identicon",
  "project_id": 4861503,
  "project": {
    "id": 4861503,
    "name": "hello-world",
    "description": "",
    "web_url": "https://gitlab.com/gitlab-org/hello-world",
    "avatar_url": null,
    "git_ssh_url": "git@gitlab.com:gitlab-org/hello-world.git",
    "git_http_url": "https://gitlab.com/gitlab-org/hello-world.git",
    "namespace": "sytses",
    "visibility_level": 0,
    "path_with_namespace": "gitlab-org/hello-world",
    "default_branch": "master",
    "ci_config_path": null,
    "homepage": "https://gitlab.com/gitlab-org/hello-world",
    "url": "git@gitlab.com:gitlab-org/hello-world.git",
    "ssh_url": "git@gitlab.com:gitlab-org/hello-world.git",
    "http_url": "https://gitlab.com/gitlab-org/hello-world.git"
  },
  "commits": [
    {
      "id": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
      "message": "added readme\n",
      "timestamp": "2017-12-10T08:26:38+00:00",
      "url": "https://gitlab.com/gitlab-org/hello-world/commit/2adc9465c4edfc33834e173fe89436a7cb899a1d",
      "author": {
        "name": "Sid Sijbrandij",
        "email": "noreply@gitlab.com"
      },
      "added": [
        "README.md"
      ],
      "modified": [
        
      ],
      "removed": [
        
      ]
    }
  ],
  "total_commits_count": 1,
  "repository": {
    "name": "hello-world",
    "url": "git@gitlab.com:gitlab-org/hello-world.git",
    "description": "",
    "homepage": "https://gitlab.com/gitlab-org/hello-world",
    "git_http_url": "https://gitlab.com/gitlab-org/hello-world.git",
    "git_ssh_url": "git@gitlab.com:gitlab-org/hello-world.git",
    "visibility_level": 0
  }
}
