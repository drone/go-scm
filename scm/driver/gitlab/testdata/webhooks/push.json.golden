{
    "Ref": "refs/heads/master",
    "Before": "9217710ce8c7e1eae7a5d1c45f6e43e1c769f866",
    "After": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
    "Repo": {
        "ID": "4861503",
        "Namespace": "gitlab-org",
        "Name": "hello-world",
        "Perm": null,
        "Branch": "master",
        "Private": false,
        "Clone": "https://gitlab.com/gitlab-org/hello-world.git",
        "CloneSSH": "git@gitlab.com:gitlab-org/hello-world.git",
        "Link": "https://gitlab.com/gitlab-org/hello-world",
        "Created": "0001-01-01T00:00:00Z",
        "Updated": "0001-01-01T00:00:00Z"
    },
    "Commit": {
        "Sha": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
        "Message": "added readme\n",
        "Author": {
            "Name": "Sid Sijbrandij",
            "Email": "noreply@gitlab.com",
            "Date": "0001-01-01T00:00:00Z",
            "Login": "sytses",
            "Avatar": "https://secure.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87?s=80\u0026d=identicon"
        },
        "Committer": {
            "Name": "Sid Sijbrandij",
            "Email": "noreply@gitlab.com",
            "Date": "0001-01-01T00:00:00Z",
            "Login": "sytses",
            "Avatar": "https://secure.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87?s=80\u0026d=identicon"
        },
        "Link": "https://gitlab.com/gitlab-org/hello-world/commit/2adc9465c4edfc33834e173fe89436a7cb899a1d"
    },
    "Commits": [
        {
            "Sha": "2adc9465c4edfc33834e173fe89436a7cb899a1d",
            "Message": "added readme\n",
            "Author": {
                "Name": "Sid Sijbrandij",
                "Email": "noreply@gitlab.com",
                "Date": "2017-12-10T08:26:38Z",
                "Login": "",
                "Avatar": ""
            },
            "Committer": {
                "Name": "Sid Sijbrandij",
                "Email": "noreply@gitlab.com",
                "Date": "2017-12-10T08:26:38Z",
                "Login": "",
                "Avatar": ""
            },
            "Link": "https://gitlab.com/gitlab-org/hello-world/commit/2adc9465c4edfc33834e173fe89436a7cb899a1d"
        }
    ],
    "Sender": {
        "Login": "sytses",
        "Name": "Sid Sijbrandij",
        "Email": "noreply@gitlab.com",
        "Avatar": "https://secure.gravatar.com/avatar/8c58a0be77ee441bb8f8595b7f1b4e87?s=80\u0026d=identicon"
    }
}